src/code/hoc/animate-spring.js

```diff
@@ -16,16 +16,24 @@ import {Motion, spring} from 'react-motion';
     in its target state without further animation. This allows animations at rest to
     respond immediately to window resize events, for instance.
  */
-export default function animateSpring(stiffness=120, damping=25) {
+export default function animateSpring(stiffness=115, damping=18) {
 
   return function(WrappedComponent) {
     return class extends React.Component {
-      
+      componentDidMount() {
+        this.fixedAnimationDuration = window.setInterval(() => {
+          if (!this.isComplete) {
+            this.handleRest();
+          }
+          window.clearInterval(this.fixedAnimationDuration);
+        }, 500);
```

Comment:
The original goal of `animateSpring()` was to be a reusable HOC appropriate for any project using `react-motion`. I think offering the option of a fixed timeout is a nice enhancement, but my predilection would be to make this a parameter so that clients would get the default `react-motion` behavior by default, but could specify an appropriate timeout when desired. Having three default parameters would be inconvenient, so switching to an options argument, e.g. `interface ISpringOptions { stiffness?: number, damping?: number, timeout?: number }` might make the most sense.